src/PowerShell/PowerShellManager.cs

```diff
@@ -212,6 +218,12 @@ internal void InvokeProfile(string profilePath)
                 durableFunctionsUtils.InitializeBindings(inputData, out bool hasExternalDFsdk);
                 Logger.Log(isUserOnlyLog: false, LogLevel.Trace, String.Format(PowerShellWorkerStrings.UtilizingExternalDurableSDK, hasExternalDFsdk));
 
+                ImportEntryPointModule(functionInfo);
+
+                // This cmdlet must be run in the same .InvokeAndClearCommands() run as the function code itself
+                // If it is not, the Activity.Current context that we are trying to preserve is dropped
+                _openTelemetryController.AddStartOpenTelemetryInvocationCommand(otelContext);
```

Comment:
I wonder if we can invoke this before `ImportEntryPointModule`. The module may contain initialization code that runs on module import, and we should capture logs that this code emits, if possible.

Reply:
Perhaps. The old worker logic was composed of two separate calls to .InvokeAndClearCommands() - one for import the entry point module and one for the customer code. 
The call to StartOpenTelemetryInvocation must happen in the same .InvokeAndClearCommands() call as the customer code, so that the internal span is generated with the same thread ID 
We could theoretically change this logic so that all three things happen in the same thread with the same .InvokeAndClearCommands() call, but this might cause unintended changes to execution logic - for instance, if Import-Module returns some value that would be passed on in the pipeline.

Reply:
Added a comment to the "Instrument profile.ps1" issue in the module repo that we should also ensure that ipmo for function definition file is also instrumented
https://github.com/Azure/azure-functions-powershell-opentelemetry/issues/12

Reply:
Please keep in mind that .InvokeAndClearCommands() does not create threads and does not make commands run in a separate thread. All the commands will run in the same thread that invokes .InvokeAndClearCommands(). This code:
```
pwsh.AddCommand("A");
pwsh.InvokeAndClearCommands();
pwsh.AddCommand("B");
pwsh.InvokeAndClearCommands();
```
is equivalent to this (from the _threading_ perspective):
```
pwsh.AddCommand("A");
pwsh.AddCommand("B");
pwsh.InvokeAndClearCommands();
```
Commands "A" and "B" will run with the same thread ID, and it will be the same thread that invokes AddCommand and InvokeAndClearCommands. With this in mind, I don't think we must follow this requirement: "The call to StartOpenTelemetryInvocation must happen in the same .InvokeAndClearCommands() call as the customer code".